quixstreams/platforms/quix/config.py

```diff
@@ -321,12 +323,43 @@ def _create_topic(self, topic: Topic, timeout: Optional[float] = None):
             cleanup_policy=cfg.extra_config.get("cleanup.policy"),
             timeout=timeout if timeout is not None else self._timeout,
         )
-        logger.info(
-            f"Creation of topic {topic_name} acknowledged by broker. Must wait "
+        logger.debug(
+            f"Creation of topic {topic.name} acknowledged by broker. Must wait "
             f"for 'Ready' status before topic is actually available"
         )
+        return resp
 
-    def _finalize_create(
+    def create_topic_no_status_check(
+        self,
+        topic: Topic,
+        timeout: Optional[float] = None,
+    ) -> dict:
+        """
+        Create topics in a Quix cluster as part of initializing the Topic object to
+        obtain the true topic name.
+
+        If it already exists, will instead get the actual topic name.
+
+        Creation is done first to avoid issues around multiple apps attempting to
+        create a topic at the same time.
+
+        :param topic: a `Topic` object
+        :param timeout: response timeout (seconds); Default 30
+        marked as "Ready" (and thus ready to produce to/consume from).
+        """
+        try:
+            return self.get_topic(topic_name=topic.name, timeout=timeout)
+        except QuixApiRequestFailure as get_error:
+            if get_error.status_code == 404:
+                try:
+                    return self._create_topic(topic, timeout=timeout)
+                except QuixApiRequestFailure as create_error:
+                    if create_error.status_code == 404:
```

Comment:
Are you sure about this ? Create fails with a 404 when the topic already exist ?

Reply:
Technically it's just a 400 with creation if it already exists (it IS a 404 with GET topic though if it doesn't exist)....pretty sure it returns a 400 with a failure message for anything BUT a successful topic creation (authentication would still be a 403 though).

I opted to no longer inspect it if it raises an error and just try the calls regardless. The stack trace will include ALL the exceptions if all attempted calls fail so the information will manifest correctly regardless, and then we're not dependent on inspecting error messages correctly or checking status codes.

Reply:
I think testing for `404` on the first get make sense. We only want to create on a 404.

Any error on create should be raised, we can't recover and do anything about it.

Reply:
Well, with creating, it might get a 404 on it because another app tried to create the topic at the same time, which is why I do the final GET attempt. 

I could still do a 404 check on the first get which will catch things like a 403 without making extra unnecc calls